### PR TITLE
build(release): add Makefile and split release script

### PR DESCRIPTION
**Pull request by scbrickley**

### CODE REVIEW

**Review by nathanielc — COMMENTED** [review]
tag-release.sh looks great, I have a question on the bump-version.sh script.

File: bump-version.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+fi
+
+release_type=$1
+if [[ $release_type != "patch" && $release_type != "minor" ]]; then
+	echo "Invalid argument: $release_type"
+	exit 1
+fi
+
+git pull origin master
+git checkout -B bump-version
```
[the remark this answers is not in the document]
Comment (scbrickley) [reply bump-version.sh:37]: I like the idea of a versioned branch name, and yes I'll remove the `git pull`

[the remark this answers is not in the document]
Comment (scbrickley) [reply bump-version.sh:37]: Do you think it's worth while to include a prompt that will inform the user of what the script will do before actually running the script? Something like:

[the remark this answers is not in the document]
Comment (scbrickley) [reply bump-version.sh:37]: Maybe with a more descriptive message, but just something to make sure the user knows what the script is doing, and give them a chance to cancel it if necessary.

[the remark this answers is not in the document]
Comment (nathanielc) [reply bump-version.sh:37]: I don't think the prompt is necessary, docs should be good enough.

[the remark this answers is not in the document]
Comment (scbrickley) [reply bump-version.sh:37]: Gotcha. Changes will be up shortly

**Review by nathanielc — APPROVED** [review]

**Pull request merged**